lib/textmeasure/textmeasure.go

```diff
@@ -138,7 +139,12 @@ func NewRuler() (*Ruler, error) {
 func (r *Ruler) addFontSize(font d2fonts.Font) {
 	sizeless := font
 	sizeless.Size = 0
-	face := truetype.NewFace(r.ttfs[sizeless], &truetype.Options{
+	ttf := r.ttfs[sizeless]
+	if ttf == nil {
+		panic(fmt.Errorf("sizeless font %s not found with style %s", font.Family, font.Style))
```

Comment:
alternatively, could pass this error up a bunch of layers I guess?

Reply:
we should instead just check earlier on for compatibility between the specified font and what the Ruler has loaded. 

Perhaps here: https://github.com/terrastruct/d2/blob/13d202b84b10468f2a4748dbed3c498592287a59/d2graph/d2graph.go#L1080

if ruler not nil and fontfamily not nil, check that ruler has fontfamily loaded.

Reply:
sounds good. i introduced a new method on `Ruler` to do this, hopefully that's what you meant? otherwise we could expose `Ruler.ttfs` but that seems a bit more strange to me.

let me know if you prefer another approach here